rpc/src/v1/types/filter.rs

```diff
@@ -78,9 +81,17 @@ impl Into<EthFilter> for Filter {
 			BlockNumber::Latest | BlockNumber::Pending => BlockId::Latest,
 		};
 
+		let (from_block, to_block) = match self.block_hash {
```

Comment:
IMHO it would be good to return a RPC error in case `block_hash`  and `from_block &/ to_block` is specified.